packages/shared/src/parsers.ts

```diff
@@ -70,7 +70,7 @@ export const ethereumAddress = hexString.refine((address) =>
 export const u64 = numericString.transform((arg) => BigInt(arg));
 
 export const u128 = z
-  .union([numericString, hexString])
+  .union([number, z.union([numericString, hexString])])
```

Comment:
wouldn't this work? 🤔 
```suggestion
  .union([number, numericString, hexString])
```

Reply:
true